pyiron_base/job/generic.py

```diff
@@ -1036,28 +1037,31 @@ def create_job(self, job_type, job_name, delete_existing_job=False):
         job._init_child_job(self)
         return job
 
-    def update_master(self):
+    def update_master(self, force_update=False):
         """
         After a job is finished it checks whether it is linked to any metajob - meaning the master ID is pointing to
         this jobs job ID. If this is the case and the master job is in status suspended - the child wakes up the master
         job, sets the status to refresh and execute run on the master job. During the execution the master job is set to
         status refresh. If another child calls update_master, while the master is in refresh the status of the master is
         set to busy and if the master is in status busy at the end of the update_master process another update is
         triggered.
+
+        Args:
+            force_update (bool): Whether to check run mode for updating master
         """
         master_id = self.master_id
         project = self.project
         self._logger.info("update master: {} {} {}".format(master_id, self.get_job_id(), self.server.run_mode))
-        if (
-            master_id is not None
-            and not self.server.run_mode.thread
-            and not self.server.run_mode.modal
-            and not self.server.run_mode.interactive
-        ):
-            self._reload_update_master(
-                project=project,
-                master_id=master_id
-            )
+        if master_id is not None:
+            if force_update or (
```

Comment:
Why do you need two if statements?

Reply:
Readability. I find separate if statements easier than one encapsulated one. But I can also put it in one if-statement if you prefer.